Source/OrchBase/P_builder.cpp

```diff
@@ -1382,7 +1382,14 @@ unsigned P_builder::WriteThreadVars(string& task_dir, unsigned coreNum,
         inPinStateInitialiser << "{";
         
         pdigraph<unsigned int, P_device*, unsigned int, P_message*, unsigned int, P_pin*>::TPp_it next_pin = (*device)->par->G.index_n[(*device)->idx].fani.upper_bound(pin_num << PIN_POS | 0xFFFFFFFF >> (32-PIN_POS));
-        for (pdigraph<unsigned int, P_device*, unsigned int, P_message*, unsigned int, P_pin*>::TPp_it p_edge = (*device)->par->G.index_n[(*device)->idx].fani.lower_bound(pin_num << PIN_POS); p_edge != next_pin; p_edge++)
+	// ADR 10 December 2019: A hack here to get the names of edge properties and state arrays the same in initialiser: the

```

Comment:
It looks like the white space at the start of the lines you are adding is not consistent with the rest of the file - please can you fix as it would be good if we were not undoing the cleanup I did in #65.

IIRC, I used two spaces instead of 4 in `P_builder.cpp` as a stop-gap until one of us got around to a full refactor.

Reply:
If anyone can indicate a *RELIABLE* way to determine existence of spaces/tabs in an application-neutral setting, without resorting to a character search, I would be very interested to know. I detest tabs yet they seem very difficult to eliminate - and there seem to be a cadre of style Nazis who insist that tabs are the required way to add indentation whitespace.

Much like Word, or OpenOffice, where things are so bad that if you even try to insert a single space at the beginning of a line, they don't even insert the additional space at all, you either have to have a linebreak or a tab. Where did this heterodoxy come from and who ever thought it would be a good idea?

Reply:
Search-and-replace is the only reliable cross-platform way. A lot of editors have plugins/options for "fixing" indentation in a file or a region. *Nix even has [indent()](https://linux.die.net/man/1/indent) and I know there are a few decrustify/[uncrustiy](https://github.com/uncrustify/uncrustify) tools about.

A lot of it comes down to editor configuration though - if your editor is configured to insert tabs rather than spaces, then your editor is sabotaging you behind your back. I certainly had to change the default setting in Notepad++ on Windows.

In a world of variable-width fonts (e.g. Word/Open Office) or where you want to do alignments, tabs can have a place.

Reply:
@AlexRast, it's a coreutils-ism, but `expand --initial --tabs=4 FILE` performs replacement (not determining existence).

Or, as @heliosfa  says, you can add this to your `.emacs`, in an appropriate editing mode or globally:

```
(setq-default indent-tabs-mode nil)
(setq-default tab-width 4)
```